src/inference/hmc.jl

```diff
@@ -337,23 +337,27 @@ function AbstractMCMC.step!(
     spl.state.eval_num = 0
 
     Turing.DEBUG && @debug "current ϵ: $ϵ"
-    
+
     # Gibbs component specified cares
     if spl.selector.tag != :default
         # Transform the space
         Turing.DEBUG && @debug "X-> R..."
         link!(spl.state.vi, spl)
         runmodel!(model, spl.state.vi, spl)
+        # Get position and log density before transition
```

Comment:
Why do we move these two lines into the if statement with duplication? Seems that we can just move them ahead of the if statement.

Reply:
I am avoiding calling `vi[spl]` more than once in the first branch.

Reply:
Alternatively,
```julia
    if spl.selector.tag != :default
        # Transform the space
        Turing.DEBUG && @debug "X-> R..."
        link!(spl.state.vi, spl)
        runmodel!(model, spl.state.vi, spl)
    end
    # Get position and log density before transition
    θ_old, log_density_old = spl.state.vi[spl], getlogp(spl.state.vi)
    if spl.selector.tag != :default
        # Update Hamiltonian
        metric = gen_metric(length(θ_old), spl)
        ∂logπ∂θ = gen_∂logπ∂θ(spl.state.vi, spl, model)
        logπ = gen_logπ(spl.state.vi, spl, model)
        spl.state.h = AHMC.Hamiltonian(metric, logπ, ∂logπ∂θ)
        resize!(spl.state.z.θ, length(θ_old))
        spl.state.z.θ .= θ_old
    end
```

Reply:
I feel this version is better as it avoids a duplicate line. How do you think?